src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -2894,13 +2901,20 @@ namespace ZoneEquipmentManager {
 
 					// initialize sizing conditions if they have not been set (i.e., no corresponding load) to zone condition
 					if ( FinalZoneSizing( CtrlZoneNum ).ZoneTempAtHeatPeak == 0.0 ) {
-						FinalZoneSizing( CtrlZoneNum ).ZoneTempAtHeatPeak = ZoneSizing( DDNumF, CtrlZoneNum ).DesHeatSetPtSeq( TimeStepAtPeakF );
+						// issue 6006, heating coils sizing to 0 when no heating load in zone
+						if ( ZoneSizing( DDNumF, CtrlZoneNum ).DesHeatSetPtSeq.empty() ) {
+							FinalZoneSizing( CtrlZoneNum ).ZoneTempAtHeatPeak = ZoneSizing( DDNumF, CtrlZoneNum ).DesHeatSetPtSeq( TimeStepAtPeakF );
```

Comment:
I did tell Mark that our arrays are never empty! My own belief was not enough to keep me from including these checks just in case.